### PR TITLE
7/debug logger

### PR DESCRIPTION
**Pull request by punkrokk**

### CODE REVIEW

File: extract_msg/message.py [hunk 1]
```diff
@@ -33,9 +37,9 @@ def __init__(self, path, prefix='', attachmentClass=Attachment, filename=None):
         :param filename: optional, the filename to be used by default when saving.
         """
         # WARNING DO NOT MANUALLY MODIFY PREFIX. Let the program set it.
-        if debug:
+        if debug._debug:
```
Comment (punkrokk) [inline extract_msg/message.py]: @TheElementalOfCreation The if debug is redundant. logger.debug will only fire if the log level is set to debug.

File: extract_msg/debug.py [hunk 2]
```diff
@@ -3,4 +3,105 @@
...
+import sys
+
+
+
+logger = logging.getLogger(__name__)
+logger.addHandler(logging.NullHandler())
+
+_debug = False
+
+def getContFileDir(_file_):
```
Comment (punkrokk) [inline extract_msg/debug.py:18]: does this belong in here? Or in the utils file?

File: extract_msg/debug.py [hunk 3]
```diff
@@ -3,4 +3,105 @@
...
+import logging.config
+import os
+import sys
+
+
+
+logger = logging.getLogger(__name__)
+logger.addHandler(logging.NullHandler())
+
+_debug = False
```
Comment (punkrokk) [inline extract_msg/debug.py:16]: Wouldn't we start with `_debug=None`? That way it actually needs to get set? And we know it was set?

File: extract_msg/debug.py [hunk 4]
```diff
@@ -3,4 +3,105 @@
...
+
+def getContFileDir(_file_):
+    """
+    Takes in the path to a file and tries to return the containing folder.
+    """
+    return '/'.join(_file_.replace('\\', '/').split('/')[:-1])
+
+def toggle_debug():
```
Comment (punkrokk) [inline extract_msg/debug.py:24]: So - my opinion is all the _debug is actually alot of extra code for no better outcome. You could just have --debug and/or see if the env var DEBUG=1. I'm still not quite sure I understand the need ( assuming in an open source repo - any one can take it and do whatever they like, including overriding your desire for them not to set debug.)

if you just read --debug and the env var of DEBUG, you could simply have a if statement in __init__.py that detects if logger level should be set to debug. (and could do the same for warning.

File: extract_msg/named.py [hunk 5]
```diff
@@ -1,7 +1,55 @@
+import logging
 from extract_msg import constants
-from extract_msg.debug import debug
+from extract_msg import debug
+from extract_msg.utils import divide#, round_up
+
+
+
+logger = logging.getLogger(__name__)
```
Comment (punkrokk) [inline extract_msg/named.py:8]: seems like logger isn't used in here... do you plan to use it in the future?

File: extract_msg/recipient.py [hunk 6]
```diff
@@ -1,18 +1,22 @@
+import logging
 from extract_msg import constants
-from extract_msg.debug import debug
+from extract_msg import debug
 from extract_msg.properties import Properties
 
 
 
+logger = logging.getLogger(__name__)
```
Comment (punkrokk) [inline extract_msg/recipient.py:8]: don't see logging used in this file either.

File: extract_msg/debug.py [hunk 7]
```diff
@@ -3,4 +3,105 @@
...
+    global _debug
+    _debug = bool(_debug ^ 1)
+    if _debug:
+        logging.getLogger().setLevel(logging.DEBUG)
+    else:
+        logging.getLogger().setLevel(logging.WARNING)
```
Comment (punkrokk) [inline extract_msg/debug.py:30]: The set level to INFO will return warning messages. You don't need to set logging.WARNING, **unless** you don't wish to show info. My guess is that isn't the case. 

This page show you how logging values work. Whatever level you set, all the values higher will alway fire.
https://docs.python.org/3/library/logging.html#levels

Also - by setting to logging.WARNING, you are turning off all the logger.info messages. 

Is that your intention?

**Comment by TheElementalOfDestruction** [comment]
@punkrokk Can you handle the conflict? I also pushed out a new commit to the final version of it

File: extract_msg/debug.py [hunk 8]
```diff
@@ -3,4 +3,105 @@
...
+    Args:
+        default_path (str): Default path to use for the logging configuration file
+        default_level (int): Default logging level
+        env_key (str): Environment variable name to search for, for setting logfile path
+
+    Returns:
+        bool: True if the configuration file was found and applied, False otherwise
+    """
+    shipped_config = getContFileDir(__file__) + '/logging-config/'
+    if os.name == 'nt':
```
Comment (punkrokk) [inline extract_msg/debug.py:49]: What Windows versions return 'nt'? All of them? Does this work across all types of windows? 

I'd recommend for *NIX (e.g. OSX and Linux) you do os.platform and look for linux and darwin. There are cases where OSX and Linux do things differently. I also know that os.platform "should" return Windows across versions like 2012, 2016, 7, 10, etc...

**Pull request merged**